### PR TITLE
Remove ImproveOSM

### LINKED ISSUE #10302: ImproveOSM service deprecated
### Description

The ImproveOSM project has been phased out. Consequently, the URLs for the publicly accessible ImproveOSM services are no longer active:
- Missing Geometry Service: previously available at https://community.improveosm.org/missingGeoService/
- Direction of Flow Service: previously available at https://community.improveosm.org/directionOfFlowService/
- Turn Restriction Service: previously available at https://community.improveosm.org/directionOfFlowService/
- 
We kindly request the removal of ImproveOSM issues map data from the iD editor.

Thank you.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by arch0345**
Closes #10302

### CODE REVIEW

File: css/20_map.css [hunk 1]
```diff
@@ -34,9 +34,6 @@
 .data-layer.osm *,
 .data-layer.notes *,
 .data-layer.keepRight *,
-.data-layer.improveOSM * {
-    pointer-events: none;
```
Comment (tordans) [inline css/20_map.css:38]: The pointer event need to stay, don't they?

File: test/spec/svg/layers.js
Comment (k-yle) [inline test/spec/svg/layers.js:1]:
```diff
-        expect(nodes.length).to.eql(18);
+        expect(nodes.length).to.eql(17);
```
to fix the failing test

**Comment by westnordost** [comment]
(Please note the last comment in https://github.com/openstreetmap/iD/issues/10302)

**Review by tyrasd — APPROVED** [review]

**Review by tyrasd — APPROVED** [review]
It's unfortunate that this service is gone, but we have to accept it. The still available partial dumps mentioned in https://github.com/openstreetmap/iD/issues/10302#issuecomment-2216837612 are not really suitable for an editor like iD.

**Pull request merged**